File: src/main/java/me/devsnox/battlekits/kits/frames/editor/KitEditorEditKitItemsFrame.java
package me.devsnox.battlekits.kits.frames.editor;

import de.tr7zw.itemnbtapi.NBTItem;
import me.devsnox.battlekits.KitManager;
import me.devsnox.battlekits.inventory.Frame;
import me.devsnox.battlekits.inventory.FrameOptions;
import me.devsnox.battlekits.kits.kit.BattleKit;
import me.devsnox.battlekits.kits.kit.KitType;
import me.devsnox.battlekits.utils.ItemBuilder;
import org.bukkit.Material;
import org.bukkit.entity.Player;
import org.bukkit.event.inventory.InventoryClickEvent;
import org.bukkit.event.inventory.InventoryCloseEvent;
import org.bukkit.inventory.ItemStack;

import java.util.ArrayList;

public class KitEditorEditKitItemsFrame extends Frame {

    private KitManager kitManager;

    private KitEditorViewFrame kitEditorViewFrame;
    private BattleKit battleKit;

    public KitEditorEditKitItemsFrame(KitManager kitManager, KitEditorViewFrame kitEditorViewFrame, BattleKit battleKit) {
        super(45, "§a§lEditor: " + battleKit.getKitType().getColor() + "§l" + battleKit.getName(), new FrameOptions(false));

        this.kitManager = kitManager;

        this.kitEditorViewFrame = kitEditorViewFrame;
        this.battleKit = battleKit;
    }

    @Override
    public void render(Player player) {

        fill(7);

        ItemBuilder itemBuilder = new ItemBuilder();

        itemBuilder.setType(battleKit.getIcon().getType());

        itemBuilder.setDisplayName(battleKit.getKitType().getColor() + battleKit.getName());
        itemBuilder.addLoreLine(battleKit.getDescription());

        this.inventory.setItem(4, itemBuilder.build());

        itemBuilder.setDisplayName("§cZurück");
        itemBuilder.setType(Material.REDSTONE);
        itemBuilder.setLore(new ArrayList<>());

        NBTItem nbtItem2 = new NBTItem(itemBuilder.build());

        nbtItem2.setBoolean("back", true);

        this.inventory.setItem(7, nbtItem2.getItem());

        for (int i = 18; i < 27 + 18; i++) {
            if(i - 18 < this.battleKit.getItems().size()) {
                inventory.setItem(i, this.battleKit.getItems().get(i - 18));
            } else {
                inventory.setItem(i, new ItemStack(Material.AIR));
            }
        }

        super.render(player);
    }

    @Override
    public void onClick(InventoryClickEvent event) {
        if (event.getWhoClicked() instanceof Player) {
            Player player = ((Player) event.getWhoClicked()).getPlayer();

            ItemStack itemStack = event.getCurrentItem();

            NBTItem nbtItem = new NBTItem(itemStack);

            if (nbtItem.hasKey("back")) {
                this.updateKit();
                this.kitEditorViewFrame.render(player, KitType.RARE);
            }
        }

        if(event.getSlot() < 18) {
            event.setCancelled(true);
        }

        super.onClick(event);
    }

    @Override
    public void onClose(InventoryCloseEvent event) {
        updateKit();
        super.onClose(event);
    }

    private void updateKit() {
        ArrayList<ItemStack> items = new ArrayList<>();

        for(int i = 18; i < this.inventory.getSize(); i++) {
            items.add(this.inventory.getItem(i));
        }

        battleKit.setItems(items);

        this.kitManager.updateKit(this.battleKit);
    }
}

File: spigot/src/main/kotlin/me/devsnox/battlekits/kits/frames/editor/KitEditorEditKitItemsFrame.kt
package me.devsnox.battlekits.kits.frames.editor

import de.tr7zw.itemnbtapi.NBTItem
import me.devsnox.battlekits.KitManager
import me.devsnox.battlekits.functions.backItem
import me.devsnox.battlekits.functions.filLBattle
import me.devsnox.battlekits.inventory.Frame
import me.devsnox.battlekits.kits.kit.BattleKit
import me.devsnox.battlekits.kits.kit.KitType
import net.darkdevelopers.darkbedrock.darkness.spigot.builder.item.ItemBuilder
import net.darkdevelopers.darkbedrock.darkness.spigot.functions.events.cancel
import org.bukkit.Material
import org.bukkit.entity.Player
import org.bukkit.event.inventory.InventoryClickEvent
import org.bukkit.event.inventory.InventoryCloseEvent

class KitEditorEditKitItemsFrame(
    private val kitManager: KitManager,
    private val kitEditorViewFrame: KitEditorViewFrame,
    private val battleKit: BattleKit
) : Frame(
    45,
    "§a§lEditor: ${battleKit.kitType.color}§l${battleKit.name}",
    false
) {

    override fun render(player: Player) {
        fill()
        val itemBuilder = ItemBuilder(battleKit.icon.type)
            .setDisplayName("${battleKit.kitType.color}${battleKit.name}")
            .addLore(battleKit.description)

        inventory.setItem(4, itemBuilder.build())
        inventory.setItem(7, backItem)
        inventory.filLBattle(battleKit.items)

        super.render(player)
    }

    override fun onClick(event: InventoryClickEvent) {
        val player = event.whoClicked as? Player ?: return
        val itemStack = event.currentItem ?: return
        if (itemStack.type == Material.AIR) return
        if (event.slot < 18) event.cancel()

        val nbtItem = NBTItem(itemStack)
        if (nbtItem.hasKey("back")) {
            updateKit()
            kitEditorViewFrame.render(player, KitType.RARE)
        }
        super.onClick(event)
    }

    override fun onClose(event: InventoryCloseEvent) {
        updateKit()
        super.onClose(event)
    }

    private fun updateKit() {
        battleKit.items = inventory.drop(18)
        kitManager.updateKit(battleKit)
    }

}

